src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/RoleMatchingResult.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol;
+
+import lombok.Data;
+
+@Data
+public class RoleMatchingResult {
+    boolean validDate;
+    boolean validCaseId;
+    boolean validJurisdiction;
+    boolean validRegion;
+    boolean validLocation;
+    boolean validClassification;
+
+    public boolean matchedAllValues() {
+        return isValidDate()
+            && isValidCaseId()
+            && isValidJurisdiction()
+            && isValidClassification()
+            && isValidRegion()
+            && isValidLocation();
+    }
+
+    public boolean matchedAExceptRegionAndLocation() {
+        return isValidDate()
+            && isValidCaseId()
+            && isValidJurisdiction()
+            && isValidClassification();
+    }
+}
```

Comment:
same comment for above. I wouldn't add this now. We'll have a separate ticket for this